Source/Core/Core/Boot/Boot.cpp

```diff
@@ -594,18 +520,46 @@ bool CBoot::BootUp(std::unique_ptr<BootParameters> boot)
 
     bool operator()(const BootParameters::IPL& ipl) const
     {
-      NOTICE_LOG_FMT(BOOT, "Booting GC IPL: {}", ipl.path);
-      if (!File::Exists(ipl.path))
+      const std::string path = SConfig::GetInstance().m_strBootROM;
+      NOTICE_LOG_FMT(BOOT, "Booting GC IPL: {}", path);
+
+      ExpansionInterface::CEXIIPL* exi_ipl = static_cast<ExpansionInterface::CEXIIPL*>(
+          ExpansionInterface::GetChannel(0)->GetDevice(2));
+      DEBUG_ASSERT(exi_ipl->m_device_type == ExpansionInterface::EXIDeviceType::MaskROM);
+
+      if (!exi_ipl->IsROMLoaded())
       {
         if (ipl.disc)
           PanicAlertFmtT("Cannot start the game, because the GC IPL could not be found.");
         else
-          PanicAlertFmtT("Cannot find the GC IPL.");
+          PanicAlertFmtT("Could not find the GC IPL.");
```

Comment:
Seems a bit odd that you only changed one branch of the check, and with a very minor change ("Cannot" vs. "Could not").

Any practical reason why past tense is more appropriate here than the previous present tense?

Reply:
I found it appropriate to use past tense here, since loading the IPL now takes place earlier, in ExpansionInterface::Init().